app/src/main/java/com/byoutline/kickmaterial/features/projectlist/ProjectsListFragment.kt

```diff
@@ -22,31 +21,29 @@ import com.byoutline.kickmaterial.utils.KickMaterialFragment
 import com.byoutline.kickmaterial.utils.LUtils
 import com.byoutline.kickmaterial.views.EndlessRecyclerView
 import com.byoutline.secretsauce.activities.showFragment
+import com.byoutline.secretsauce.di.Injectable
+import com.byoutline.secretsauce.di.inflateAndSetViewModel
+import com.byoutline.secretsauce.di.lazyViewModelWithAutoLifecycle
 import timber.log.Timber
-import javax.inject.Inject
+
 
 /**
  * @author Pawel Karczewski <pawel.karczewski at byoutline.com> on 2015-01-03
  */
-class ProjectsListFragment : KickMaterialFragment(), ProjectClickListener, FieldStateListener, EndlessRecyclerView.EndlessScrollListener {
-
-    @Inject
-    lateinit var viewModel: ProjectListViewModel
+class ProjectsListFragment : KickMaterialFragment(), Injectable, ProjectClickListener, FieldStateListener, EndlessRecyclerView.EndlessScrollListener {
 
+    private val viewModel by lazyViewModelWithAutoLifecycle(this, ProjectListViewModel::class)
+    lateinit var binding: FragmentProjectsBinding
 
     private lateinit var category: Category
-    private lateinit var binding: FragmentProjectsBinding
     /**
      * Endless scroll variables *
      */
     private lateinit var layoutManager: GridLayoutManager
     private lateinit var scrollListener: ProjectsListScrollListener
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        val binding = FragmentProjectsBinding.inflate(inflater, container, false)
-        this.binding = binding
-        KickMaterialApp.component.inject(this)
-        binding.viewModel = viewModel
```

Comment:
we don't have to set viewModel since `inflateAndSetViewModel` does it for us.
We do not have to inject, since `lazyViewModelWithAutoLifecycle` knows how to get viewModelFactory